File: GFG/Arrays/leadersofarray.cpp
#include<iostream>
#include<vector>
using namespace std;

void leader(vector<int> arr){
    int n= arr.size();
    int j=n-1;
    int max=INT_MIN;
    while(j>=0){
        if(max<arr[j]){
            max=arr[j];
            cout<<max<<" ";
            j--;
        }
        else{j--;}
    }

    return ;
}

int main (){
    vector<int> brr={7,10,4,3,6,5,2};
    leader(brr);
    return 0;
}

File: GFG/Arrays/leadersofarray.py
def leader(arr):
    n=len(arr)
    j=n-1
    max=-(2**31)+1
    while j>=0:
        if max<arr[j]:
            max=arr[j]
            print(max)
            j-=1
        else:j-=1
    return arr

brr=[7,10,4,3,6,5,2]
leader(brr)